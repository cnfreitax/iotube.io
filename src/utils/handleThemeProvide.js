import { themeGet } from '@styled-system/theme-get'

const getMediaQuery = (size) => (props) => {
  const queries = themeGet('mediaQueries')(props) || {}
  console.log(queries[size])
  return queries[size]
}

const getPaletteColor = (color) => (props) => {
  return themeGet('colors')(props)[color]
}

const getFontSize = (size) => (props) => {
  return themeGet('fontSize')(props)[size]
}

export { getMediaQuery, getPaletteColor, getFontSize }
