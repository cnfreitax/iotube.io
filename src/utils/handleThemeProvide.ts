import { themeGet } from '@styled-system/theme-get'
import { ColorType, FontSizeType } from 'interfaces/theme'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getMediaQuery = (size: string) => (props: any): any => {
  const queries = themeGet('mediaQueries')(props) || {}
  console.log(queries[size])
  return queries[size]
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getPaletteColor = (color: ColorType) => (props: any): string => {
  return themeGet('colors')(props)[color]
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getFontSize = (size: FontSizeType) => (props: any): string => {
  return themeGet('fontSize')(props)[size]
}

export { getMediaQuery, getPaletteColor, getFontSize }
