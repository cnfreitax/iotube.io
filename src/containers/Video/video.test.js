import { render } from 'utils/test';
import { VideoContainer } from '.';

describe('[container] Video', () => {
  it('should match snapshot', () => {
    const sut = render(<VideoContainer />)
      expect(sut).toMatchSnapshot()
  });
});
