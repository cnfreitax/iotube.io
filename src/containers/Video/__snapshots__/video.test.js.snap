// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[container] Video should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  color: #282626;
  font-size: 1.8rem;
  font-weight: 400;
  line-height: 1.5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 25px;
  background-color: #F2C185;
  border-bottom: 1px solid #dbdbdb;
  position: -webkit-sticky;
  position: sticky;
  width: 100%;
  height: 100%;
  max-height: 74px;
  top: 0;
  z-index: 10000;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  background: transparent;
  border: none;
  cursor: pointer;
}

.c0 {
  min-height: 100%;
  width: 100%;
  background-color: #F2F0EB;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-height: calc(100% - 74px);
  padding: 14px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  background-color: red;
  width: 100%;
}

<body>
    <div>
      <div
        class="c0"
      >
        <header
          class="c1"
        >
          <div
            class="c2"
          >
            <span
              class="c3"
              size="3"
            >
              nome usuario
            </span>
            <span
              class="c3"
              size="3"
            >
              email@usuar.com
            </span>
          </div>
          <button
            class="c4"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 26px; height: 26px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
            >
              <img
                data-nimg="fixed"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </button>
        </header>
        <main
          class="c5"
        >
          <div
            class="c6"
          >
            <span>
              ok
            </span>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Wrapper-ovcdzi-0 bcauVb"
    >
      <header
        class="styles__Header-sc-7r338f-0 gpFoCf"
      >
        <div
          class="styles__DivInfoUser-sc-7r338f-1 ihvxJK"
        >
          <span
            class="styles__Heading-sc-1v3nzhy-0 NiNKa"
            size="3"
          >
            nome usuario
          </span>
          <span
            class="styles__Heading-sc-1v3nzhy-0 NiNKa"
            size="3"
          >
            email@usuar.com
          </span>
        </div>
        <button
          class="styles__ButtonLogout-sc-7r338f-2 bUWCFL"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 26px; height: 26px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <img
              data-nimg="fixed"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
      </header>
      <main
        class="styles__Content-ovcdzi-1 bHUOHL"
      >
        <div
          class="styles__Wrapper-sc-17900wp-0 DttRs"
        >
          <span>
            ok
          </span>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
